sig-security-assessments/cluster-api/self-assessment.md

```diff
@@ -0,0 +1,1099 @@
+  command. Verify checksum of the downloaded components. Download with TLS
+  enabled to prevent in-transit tampering.
+  * Status: implemented / planned / to be implemented / End User Guidance
+* During runtime: Alert on modification or restarts of cluster API components to
+  detect for potential tampering.
+  * Status: [End user guidance to be written](#cluster-api-end-user-guide)
+
+#### STRIDE-TAMPER-2 Tampering of Machine Images
+
+Machine images used as Node OS images, can be tampered during build time, during
+install time and during runtime
+
+#### STRIDE-TAMPER-2 Recommended Mitigations
+
+* At build time: Create SBoM of the Machine Image and verify checksum as a post
+  build action
+  *
+  Status: [Issue filed](https://github.com/kubernetes-sigs/image-builder/issues/803)
+* During install time: Perform mutual authentication when invoking clusterctl
+  command. Verify checksum of the downloaded machine image, before marking them
+  as available. Download with TLS enabled to prevent in-transit tampering.
+  * Status: Partially implemented
+    * Maintainers note: clusterctl by default uses Github APIs for releases
+* During runtime: Disable or restrict SSH access to prevent updates to machine
+  image at runtime. Create alerting for sensitive file systems and when the
+  machine is restarted.
+  * Status: [End user guidance to be written](#cluster-api-end-user-guide)
+    * Maintainers note: No password or SSH auth key is stored in image builder
+      by default, and a user may optionally set a password or add authorised
+      keys during first-boot using cloud-init configurations.
+
+### Repudiation
+
+#### STRIDE-REPUDIATE-01 Repudiation of actions performed by Cloud Admin
+
+Infra Admin has highly privileged access to cloud and cluster resources. Misuse
+of permissions via actions performed by Infra Admin should be detectable and
+logged for auditing purposes.
+
+#### STRIDE-REPUDIATE-01 Recommended Mitigations
+
+* Implement API server auditing for all clusters and Cloud API auditing to log
+  all actions performed by Infra Admins. Additionally, implement centralized
+  audit collection and alerting for any suspicious activity.
+  * Status: [End user guidance to be written](#cluster-api-end-user-guide)
+
+#### STRIDE-REPUDIATE-02 Repudiation of actions performed using cloud credentials
+
+Cloud credentials that are used by the Cluster API components, have highly
+privileged access to cloud resources. Misuse of these credentials by accidental
+leakage or by cluster admins, should be detectable and logged for audit purposes
+
+#### STRIDE-REPUDIATE-02 Recommended mitigations
+
+* Recommended Mitigation: Implement Cloud API auditing to log all actions
+  performed using cloud credentials granted to the management and workload
+  cluster’s control plane. Additionally, implement centralized audit collection
+  and alerting for any suspicious activity.
+  * Status: [End user guidance to be written](#cluster-api-end-user-guide)
```

Comment:
I wonder if we are able to deliver that guidance or rather in which details we can provide it.

Some of the points seem to have a very large scope (I would say some ~ the same size as the whole CAPI project, e.g. it's non trivial to build/deploy/configure/integrate a full-fledged monitoring and auditing solution).